File: P04 - Inherirance/Java/Manajer.java
package Praktikum.P04.P04c_Latihan_Inheritance_Java_227008_Elmosius_Suli;
// File : Manajer.java
// Nama : Elmosius Suli
// NRP  : 2272008
// Kelas : B
// Ket :
public class Manajer extends Pegawai{
    private int bonus;

    // constructor
    public Manajer(String namanya, int gajinya){
        super(namanya, gajinya);
    }

    // method memberi nilai bonus
    public void setBonus(int bonusnya){
        this.bonus = bonusnya;
    }

    // method meminta nilai gaji
    public int getGaji(){
        int gajiPokok = super.getGaji();
        return gajiPokok + this.bonus;
    }
}

File: P04 - Inherirance/Python/Manajer.py
from Pegawai import Pegawai

class Manajer(Pegawai):
    def __init__(self, namanya: str, gajinya: int) -> None:
        super().__init__(namanya, gajinya)
        self.__bonus: int =0
        
        
    @property
    def bonus(self) -> int:
        return self.__bonus
    
    @bonus.setter
    def bonus(self, bonusnya: int) -> None:
        self.__bonus = bonusnya
        
    @property
    def gaji(self)-> int:
        gaji_pokok = super().gaji
        return gaji_pokok + self.bonus